test/verifyCredentials.spec.js

```diff
@@ -0,0 +1,71 @@
+const { expect } = require('chai');
+const logger = require('@elastic.io/component-logger')();
+const verify = require('../verifyCredentials');
+
+describe('verifyCredentials test', () => {
+  const emitter = {
+    logger,
+  };
+
+  describe('oauth2 credentials', () => {
+    const credentials = {
+      auth: {
+        type: 'OAuth2',
+        oauth2: {
+          clientId: 'clientId',
+          clientSecret: 'clientSecret',
+          authUri: 'https://example.com/oauth/v2/auth',
+          tokenUri: 'https://example.com/oauth/v2/token',
+          scopes: [
+            'Example.fullaccess.all',
+          ],
+          additionalProperties: {},
+          keys: {},
+        },
+      },
+    };
+
+    it('should succeed', async () => {
+      credentials.auth.oauth2.keys = {
+        access_token: 'access_token',
+        refresh_token: 'refresh_token',
+      };
+      const result = await verify.call(emitter, credentials);
+      expect(result).equal(true);
+    });
+
+    it('should fail if auth.oauth2.keys is missing', async () => {
+      credentials.auth.oauth2.keys = undefined;
+      try {
+        await verify.call(emitter, credentials);
+        throw new Error('This line should never be called because await above should throw an error');
+      } catch (err) {
+        expect(err.message).equal('Error: OAuth2 provider hasn`t returned keys for current credentials');
+      }
```

Comment:
Just a suggestion - instead of using `try...catch` you could also use chaiAsPromised's methods for testing errors.